src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+	 * @return
+	 */
+	protected File selectEntry(String dialogTitle) {
+
+		JFileChooser chooser = new JFileChooser(cdpTextField.getText());
+//        chooser.setFileSelectionMode(JFileChooser.FILES_ONLY);
+		FileFilter filterYAML = new FileNameExtensionFilter("YAML document (*.yaml)", "yaml");
+		chooser.addChoosableFileFilter(filterYAML);
+
+		chooser.setAcceptAllFileFilterUsed(false);
+		chooser.setFileFilter(filterYAML);
+
+		int action = chooser.showOpenDialog(this);
+		chooser.setDialogTitle(dialogTitle);
+
+		switch (action) {
+		case JFileChooser.APPROVE_OPTION:
+			File file = chooser.getSelectedFile();
+			return file;
+		case JFileChooser.CANCEL_OPTION:
+		case JFileChooser.ERROR_OPTION:
+		default:
+			break;
+		}
+		return null;
+
+	}
+
+	void updateClientCDPPath(String path) {
+		ClientSettings settings = getContest().getClientSettings();
+		if (settings == null) {
+			settings = new ClientSettings(getContest().getClientId());
+		}
+		settings.put(CUSTOM_SUBMIT_SAMPLE_CDP_PATH, path);
+
+		getController().updateClientSettings(settings);
+	}
+
+	private void xlog(String string) {
+		System.out.println(string);
+		addLineToTextArea(string);
+	}
+
+	protected String getClientCDPPath() {
+
+		ClientSettings settings = getContest().getClientSettings();
+		if (settings != null) {
+			String path = settings.getProperty(CUSTOM_SUBMIT_SAMPLE_CDP_PATH);
+			if (path != null) {
+				return path;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Reset fields back to default values
+	 * 
+	 * @param usingGui
+	 */
+	protected void resetFields(boolean isUsingGui) {
+	    
+	    
+	    selectedJudgingTypeIndexes = new int[0];
+	    selectedProblemsIndexes = new int[0];
+	    selectedLanguagesIndexes = new int[0];
+
+		checkBoxSubmitYesSamples.setSelected(true);
+		checkBoxSubmitFailingSamples.setSelected(true);
+		
+		checkBoxLanguage.setSelected(true);
+        checkBoxProblems.setSelected(true);
+        
+        submitAllJudgingTypesRadioButton.setSelected(true);
+        submitAllProblemsRadioButton.setSelected(true);
+        submitAllLanguagesRadioButton.setSelected(true);
+
+		String cdpPath = getContest().getContestInformation().getJudgeCDPBasePath();
+
+		String clientPath = getClientCDPPath();
+
+		if (clientPath != null && isUsingGui) {
+			int result = FrameUtilities.yesNoCancelDialog(this, "Overwrite locally saved CDP path with this (default) " + cdpPath,
+					"Replace CDP Path?");
+			if (result == JOptionPane.NO_OPTION) {
+				cdpPath = clientPath;
+			}
+		}
+
+		cdpTextField.setText(cdpPath);
+
+//		getController().updateClientSettings(settings);;
+
+	}
+
+	/**
+	 * Submit sample solutions.
+	 */
+	protected void submitSelectedJudgesSolutions() {
+	    showMessage("");
+
+	    String warningMessage = verifyCDP (cdpTextField.getText());
+	    if (warningMessage != null) {
+	        // let the user know that the CDP selected may not work
+	        FrameUtilities.showMessage(this, "Trying to use an invalid CDP?", warningMessage);
```

Comment:
If I was a user and this message popped up, I would be confused... I'd be saying "WHAT 'invalid CDP'?"  I think the warning message should include the value of the (allegedly) invalid CDP (in other words, include the value of `cdpTextField.getText()`).